platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/SyncGossip.java

```diff
@@ -236,13 +245,27 @@ protected SyncGossip(
         reconnectMetrics = new ReconnectMetrics(platformContext.getMetrics(), addressBook);
 
         final StateConfig stateConfig = platformContext.getConfiguration().getConfigData(StateConfig.class);
+
+        final LongSupplier getRoundSupplier = () -> {
+            try (final ReservedSignedState reservedState = latestCompleteState.get()) {
+                if (reservedState == null || reservedState.isNull()) {
+                    return ROUND_UNDEFINED;
+                }
+
+                return reservedState.get().getRound();
+            }
+        };
+
         reconnectHelper = new ReconnectHelper(
                 this::pause,
                 clearAllPipelinesForReconnect::run,
                 swirldStateManager::getConsensusState,
-                latestCompleteState::getRound,
+                getRoundSupplier,
                 new ReconnectLearnerThrottle(platformContext.getTime(), selfId, reconnectConfig),
-                loadReconnectState,
+                state -> {
+                    loadReconnectState.accept(state);
+                    resetFallenBehind();
```

Comment:
optional: replace the call to `resetFallenBehind` with

```
syncManager.resetFallenBehind();
```

Reply:
done